### ISSUE TITLE
After upgrade to 3.2 beta, clicking on AnimationPlayer node crashes editor

### ISSUE THREAD

**Issue by benbishopnz** [godotengine/godot#33897]
**Godot version:**
3.1.1 -> 3.2 beta

**OS/device including version:**
MacOS 10.14.6 Mohave

**Issue description:**
After opening my project in 3.2 beta, clicking (left or right click) upon an AnimationPlayer node crashes the editor.

**Steps to reproduce:**
I originally created the AnimationPlayer node in version 3.0, updated to 3.1 and 3.1.1 and I noticed the AnimationPlayer wasn't playing the animation after updating to 3.2 beta. When I went to check the node the editor crashes consistently.

**Comment by Calinou** [comment]
To get more information about the crash, you could [compile Godot from source](https://docs.godotengine.org/en/latest/development/compiling/compiling_for_osx.html) to get a debug build. You can then run this build from a terminal, open your project and view the crash backtrace in the terminal.

Also, could you upload your project somewhere if possible? This would greatly speed up troubleshooting.

**Comment by ctemplin** [comment]
I had this same problem with 3.2 stable. I fixed it by editing the .tscn, removing the `root_node...` line.

```
[node name="AnimationPlayer" type="AnimationPlayer" parent="."]
root_node = NodePath("")
```

**Comment by zyansheep** [comment]
I have this same problem too. Backtrace:
```ERROR: get_node: Node not found: .
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: .
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: .
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: .
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: .
   At: scene/main/node.cpp:1381.
ERROR: _ensure_node_caches: Condition "!parent" is true.
   At: scene/animation/animation_player.cpp:239.
ERROR: _animation_process_animation: Condition "p_anim->node_cache.size() != p_anim->animation->get_track_count()" is true.
   At: scene/animation/animation_player.cpp:347.
ERROR: get_node: Node not found: .
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: .
   At: scene/main/node.cpp:1381.
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /usr/lib/libc.so.6(+0x3bd70) [0x7f0269d00d70] (??:0)
[2] /usr/bin/godot() [0x1c6cca4] (??:?)
[3] /usr/bin/godot() [0x1c6da05] (??:?)
[4] /usr/bin/godot() [0x1c75441] (??:?)
[5] /usr/bin/godot() [0x1a48d88] (??:?)
[6] /usr/bin/godot() [0x1a5382e] (??:?)
[7] /usr/bin/godot() [0x1a5695b] (??:?)
[8] /usr/bin/godot() [0x2c53bf4] (??:?)
[9] /usr/bin/godot() [0x212d55d] (??:?)
[10] /usr/bin/godot() [0x834c99] (??:0)
[11] /usr/bin/godot() [0x2c59ae2] (??:?)
[12] /usr/bin/godot() [0x2c4e64b] (??:?)
[13] /usr/bin/godot() [0x2c4e928] (??:?)
[14] /usr/bin/godot() [0x1cb5a8b] (??:?)
[15] /usr/bin/godot() [0x7c4d44] (??:0)
[16] /usr/bin/godot() [0x7ab371] (??:0)
[17] /usr/bin/godot() [0x79973d] (??:0)
[18] /usr/lib/libc.so.6(__libc_start_main+0xf3) [0x7f0269cec023] (??:0)
[19] /usr/bin/godot() [0x79bf2e] (??:0)
-- END OF BACKTRACE --```

**Comment by akien-mga** [comment]
Can anyone provide a project that triggers this crash?

**Comment by zyansheep** [comment]
[Crash.zip](https://github.com/godotengine/godot/files/4465740/Crash.zip) crashes editor when clicking AnimationPlayer
Errors on attempting to find ../icon
```OpenGL ES 3.0 Renderer: Mesa Intel(R) UHD Graphics 620 (KBL GT2)
ERROR: get_node: Node not found: ../icon.
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: ../icon.
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: ../icon.
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: ../icon.
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: ../icon.
   At: scene/main/node.cpp:1381.
ERROR: _ensure_node_caches: Condition "!parent" is true.
   At: scene/animation/animation_player.cpp:239.
ERROR: _animation_process_animation: Condition "p_anim->node_cache.size() != p_anim->animation->get_track_count()" is true.
   At: scene/animation/animation_player.cpp:347.
ERROR: get_node: Node not found: ../icon.
   At: scene/main/node.cpp:1381.
ERROR: get_node: Node not found: ../icon.
   At: scene/main/node.cpp:1381.
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /usr/lib/libc.so.6(+0x3bd70) [0x7f5acbd0cd70] (??:0)
[2] /usr/bin/godot() [0x1c6cca4] (??:?)
[3] /usr/bin/godot() [0x1c6da05] (??:?)
[4] /usr/bin/godot() [0x1c75441] (??:?)
[5] /usr/bin/godot() [0x1a48d88] (??:?)
[6] /usr/bin/godot() [0x1a5382e] (??:?)
[7] /usr/bin/godot() [0x1a5695b] (??:?)
[8] /usr/bin/godot() [0x2c53bf4] (??:?)
[9] /usr/bin/godot() [0x212d55d] (??:?)
[10] /usr/bin/godot() [0x834c99] (??:0)
[11] /usr/bin/godot() [0x2c59ae2] (??:?)
[12] /usr/bin/godot() [0x2c4e64b] (??:?)
[13] /usr/bin/godot() [0x2c4e928] (??:?)
[14] /usr/bin/godot() [0x1cb5a8b] (??:?)
[15] /usr/bin/godot() [0x7c4d44] (??:0)
[16] /usr/bin/godot() [0x7ab371] (??:0)
[17] /usr/bin/godot() [0x79973d] (??:0)
[18] /usr/lib/libc.so.6(__libc_start_main+0xf3) [0x7f5acbcf8023] (??:0)
[19] /usr/bin/godot() [0x79bf2e] (??:0)
-- END OF BACKTRACE --```

**Comment by zyansheep** [comment]
More detailed backtrace with compiled 3.2 stable branch (running Linux X11)
```
[1] /usr/lib/libc.so.6(+0x3bd70) [0x7f2ceec19d70] (??:0)
[2] Node::get_node_or_null(NodePath const&) const (/home/zyansheep/Code/Godot/Git/godot/scene/main/node.cpp:1320)
[3] Node::get_node(NodePath const&) const (/home/zyansheep/Code/Godot/Git/godot/scene/main/node.cpp:1380)
[4] Node::get_node_and_resource(NodePath const&, Ref<Resource>&, Vector<StringName>&, bool) const (/home/zyansheep/Code/Godot/Git/godot/scene/main/node.cpp:2509)
[5] AnimationTrackEdit::_is_value_key_valid(Variant const&, Variant::Type&) const (/home/zyansheep/Code/Godot/Git/godot/editor/animation_track_editor.cpp:2485 (discriminator 1))
[6] AnimationTrackEdit::draw_key(int, float, int, bool, int, int) (/home/zyansheep/Code/Godot/Git/godot/editor/animation_track_editor.cpp:2280 (discriminator 1))
[7] AnimationTrackEdit::_notification(int) (/home/zyansheep/Code/Godot/Git/godot/editor/animation_track_editor.cpp:2024 (discriminator 3))
[8] AnimationTrackEdit::_notificationv(int, bool) (/home/zyansheep/Code/Godot/Git/godot/editor/animation_track_editor.h:125 (discriminator 14))
[9] Object::notification(int, bool) (/home/zyansheep/Code/Godot/Git/godot/core/object.cpp:933)
[10] CanvasItem::_update_callback() (/home/zyansheep/Code/Godot/Git/godot/scene/2d/canvas_item.cpp:460)
[11] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (/home/zyansheep/Code/Godot/Git/godot/./core/method_bind.gen.inc:61 (discriminator 4))
[12] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/zyansheep/Code/Godot/Git/godot/core/object.cpp:921 (discriminator 1))
[13] MessageQueue::_call_function(Object*, StringName const&, Variant const*, int, bool) (/home/zyansheep/Code/Godot/Git/godot/core/message_queue.cpp:250)
[14] MessageQueue::flush() (/home/zyansheep/Code/Godot/Git/godot/core/message_queue.cpp:299)
[15] SceneTree::iteration(float) (/home/zyansheep/Code/Godot/Git/godot/scene/main/scene_tree.cpp:484 (discriminator 2))
[16] Main::iteration() (/home/zyansheep/Code/Godot/Git/godot/main/main.cpp:2005)
[17] OS_X11::run() (/home/zyansheep/Code/Godot/Git/godot/platform/x11/os_x11.cpp:3257)
[18] /home/zyansheep/Code/Godot/Git/godot/bin/godot.x11.tools.64(main+0x121) [0x13f76a7] (/home/zyansheep/Code/Godot/Git/godot/platform/x11/godot_x11.cpp:57)
[19] /usr/lib/libc.so.6(__libc_start_main+0xf3) [0x7f2ceec05023] (??:0)
[20] /home/zyansheep/Code/Godot/Git/godot/bin/godot.x11.tools.64(_start+0x2e) [0x13f74ce] (??:?)
-- END OF BACKTRACE --
[1]    79094 abort (core dumped)  /home/zyansheep/Code/Godot/Git/godot/bin/godot.x11.tools.64 -e

**Comment by KoBeWi** [comment]
Seems fixed in 3.3 (tested with the attached project)
